File: backend/src/Spisa.Domain/Entities/Category.cs
using Spisa.Domain.Common;

namespace Spisa.Domain.Entities;

public class Category : BaseEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public decimal DefaultDiscountPercent { get; set; } = 0;

    // Navigation properties
    public ICollection<Article> Articles { get; set; } = new List<Article>();
    public ICollection<ClientDiscount> ClientDiscounts { get; set; } = new List<ClientDiscount>();
}

File: frontend/types/category.ts
export interface Category {
  id: number;
  code: string;
  name: string;
  description?: string;
  defaultDiscountPercent: number;
  isDeleted: boolean;
  createdAt: string;
  updatedAt: string;
  articlesCount: number;
}

export interface CategoryFormData {
  code: string;
  name: string;
  description?: string;
  defaultDiscountPercent: number;
}







